packages/neuron-ui/src/containers/Main/hooks.ts

```diff
@@ -27,34 +27,45 @@ const SYNC_INTERVAL_TIME = 4000
 const CONNECTING_BUFFER = 15_000
 let CONNECTING_DEADLINE = Date.now() + CONNECTING_BUFFER
 
+const isCurrentUrl = (url: string) => {
+  const id = currentNetworkIDCache.load()
+  const list = networksCache.load()
+  const cached = list.find(n => n.id === id)?.remote
+  return cached === url
+}
+
 export const useSyncChainData = ({ chainURL, dispatch }: { chainURL: string; dispatch: StateDispatch }) => {
   useEffect(() => {
     let timer: NodeJS.Timeout
     const syncBlockchainInfo = () => {
       Promise.all([getTipHeader(), getBlockchainInfo()])
         .then(([header, chainInfo]) => {
-          dispatch({
-            type: AppActions.UpdateChainInfo,
-            payload: {
-              tipBlockNumber: `${BigInt(header.number)}`,
-              tipBlockHash: header.hash,
-              tipBlockTimestamp: +header.timestamp,
-              chain: chainInfo.chain,
-              difficulty: BigInt(chainInfo.difficulty),
-              epoch: chainInfo.epoch,
-            },
-          })
+          if (isCurrentUrl(chainURL)) {
```

Comment:
I am curious to know bit more about the scenarios where `chainURL` is not equal to `current url`. 
Can you please clarify a bit when the `isCurrentUrl(chainURL) === false`?

Reply:
Suppose that the request is sent to `url a` but the response is held, while the network config is set to  `url b`. Then the response from `url a` should be discarded.